proxy/http/HttpTransactCache.cc

```diff
@@ -1299,7 +1299,7 @@ HttpTransactCache::match_response_to_request_conditionals(HTTPHdr *request, HTTP
       ink_time_t lm_value = response->get_last_modified();
 
       // we won't return NOT_MODIFIED if Last-modified is too recent
-      if ((lm_value == 0) || (request->get_if_modified_since() < lm_value)) {
```

Comment:
The logic that exist in ATS is correct it should pass back a 304 if the last modified and if-modified-since match.  This is how HTTP/1.1 clients should work.

This change would break everyone else.  I will comment more in the bug TS-3995.